docs/rfc/rfc-1.adoc

```diff
@@ -633,13 +633,138 @@ As time passes and operators drop out of the system, a wallet becomes at risk
+chain declaring an intention to close the wallet.
+
+That operator proposes a BTC fee (as in <<sweeping,sweeping>>) and the
+operators use the wallet's public key hash to choose viable (it must also have
+a passing heartbeat and be above `wallet_min_btc`, and not exceeded the
+`wallet_max_age) wallet(s) to transfer the remaining funds to. They construct a
+P2PKH transaction moving the wallet's main UTXO to that wallet. For more
+details on exactly how a wallet is chosen and how the funds are split up see
+<<wallet-closure-transaction,the dedicated section>>.
+
+After this transaction is complete, <<public-knowledge,the public>> is able to
+submit a SPV proof to let ethereum know that the funds were transferred
+nonfraudulently. Later, a <<donate,donation>> can be made by governance to
+handle any difference in outstanding v2 token supply and locked BTC due to
+mining transfer fees.
+
+*Notes*: Transferring the BTC to any address other than the P2PKH of one of the
+other wallet addresses is fraud, and is <<Punishment,punishable>>. Moving funds
+without sending a follow-up proof is <<Punishment,punishable>>. Failure to
+close a wallet that failed a <<heartbeat,heartbeat>> or fell below
+`wallet_min_btc`, or exceeded `wallet_max_age` is <<Punishment,punishable>>.
+Failure to complete the process before `wallet_closure_timeout` has elapsed is
+<<Punishment,punishable>>.
+
+Any unswept funds can be returned via the 30-day return script, though since
+we're redeeming from the oldest wallet and only sweeping/depositing to the
+newest wallets this hopefully won't come up often 🤞.
+
+[[wallet-closure-transaction]]
+===== Wallet Closure Transactions
+
+Governable Parameters:
+
+- `wallet_transfer_max`: The most amount of BTC a wallet can transfer to a
+  single other wallet during a wallet closure.
+
+When a wallet <<closing-a-wallet,closes>>, if there is any amount of bitcoin
+remaining, it needs to be transfered to another _live_ wallet. Our input to the
+transaction is the closing wallet's main UTXO, and we create an output for
+every multiple of `wallet_transfer_max` we have in BTC in the closing wallet.
+
+The destination wallets are chosen for their public key hash's modulus distance
+to the closing wallet's public key hash (both represented as a number). Ties are
+broken by the destination with the greater public key hash, and if we run out of
+viable wallets, it is okay to reuse them (though we prefer not to).
+
+For all the below examples, say that the public key hashes can range from 0 to
+99 (mod 100). Wallet#43 Means that the PKH of the Wallet is 43. Real public key
+hashes and their max mod values will be much higher in practice!
+
+Exceeding Transfer Max Example:
+
+- Wallet#90 with balance `250 BTC` needs to close.
+- `wallet_transfer_max = 100 BTC`
+- Live wallet options are Wallet#30, Wallet#80, Wallet#25, Wallet#15
+- Calculate distances:
+- Wallet#30 = min(90 - 30, 130 - 90) = min(60, 40) = 40
```

Comment:
We have two arguments passed to `min` because we want to look from both sides, right? That is, we can imagine the list of wallets as a circle and not as a line. Is it needed though? 

Thinking of it just as just a line would be a bit easier to implement. Are there reasons for doing it another way?

Reply:
Yup! Explained it in depth [above](https://github.com/keep-network/tbtc-v2/pull/127#discussion_r828731782)